services/src/tpu_utils/quinn_auto_reconnect.rs

```diff
@@ -0,0 +1,213 @@
+use anyhow::{bail, Context};
+use log::{info, warn};
+use quinn::{Connection, ConnectionError, Endpoint};
+use std::fmt;
+use std::net::SocketAddr;
+use std::time::Duration;
+use tokio::sync::RwLock;
```

Comment:
Seems similar code, should be benchmarked with existing code, I prefer only to have one way that we connect and maintain connection with TPU client. Could you compare the results of existing implementation and this implementation. You can also refactor original implementation so that it reflects the changes you want to do.